File: problems/3314-construct-the-minimum-bitwise-array-i/solution.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> ans(nums.size());
        for (int i = 0; i < nums.size(); ++i) {
            int num = nums[i];
            int x = num - 1;
            if ((x | (x + 1)) == num) {
                ans[i] = x;
            } else {
                ans[i] = -1;
            }
        }
        return ans;
    }
};

File: problems/3314-construct-the-minimum-bitwise-array-i/solution.java
import java.util.*;
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int n = nums.size();
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            int num = nums.get(i);
            int x = -1;
            for (int j = 0; j < num; j++) {
                if ((j | (j + 1)) == num) {
                    x = j;
                    break;
                }
            }
            ans[i] = x;
        }
        return ans;
    }
}